src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -381,79 +558,13 @@ osx_render(void *vdata,
 	   UInt32 in_number_frames,
 	   AudioBufferList *buffer_list)
 {
-	AudioBuffer *output_buffer = nullptr;
-	size_t output_buffer_frame_size;
-
 	OSXOutput *od = (OSXOutput *) vdata;
 
-	/*
-		By convention when interfacing with audio hardware in CoreAudio,
-		in_bus_number equals 0 for output and 1 for input. Because this is an
-		audio output plugin our in_bus_number should always be 0.
-	*/
-	assert(in_bus_number == 0);
-
-	unsigned int input_channel_count = od->asbd.mChannelsPerFrame;
-	unsigned int output_channel_count = 0;
-	for (unsigned int i = 0 ; i < buffer_list->mNumberBuffers; ++i) {
-		output_buffer = &buffer_list->mBuffers[i];
-		assert(output_buffer->mData != nullptr);
-		output_channel_count += output_buffer->mNumberChannels;
-	}
-	assert(output_channel_count == input_channel_count);
-
-	size_t input_buffer_frame_size = od->asbd.mBytesPerFrame;
-	size_t sample_size = input_buffer_frame_size / input_channel_count;
-
-	size_t requested_bytes = in_number_frames * input_buffer_frame_size;
-	if (requested_bytes > od->render_buffer_size)
-		requested_bytes = od->render_buffer_size;
-
-	size_t available_bytes = od->ring_buffer->pop(od->render_buffer, requested_bytes);
-
-	/*
-		Maybe this is paranoid but we have no way of knowing
-		if the 'pop' above ended at a frame boundary. In case
-		of an incomplete last frame, keep popping until the
-		last frame is complete.
-	*/
-	while (true) {
-		size_t incomplete_frame_bytes = available_bytes % input_buffer_frame_size;
-		if (incomplete_frame_bytes == 0)
-			break;
-
-		available_bytes += od->ring_buffer->pop(
-			od->render_buffer + available_bytes,
-			input_buffer_frame_size - incomplete_frame_bytes
-		);
-	}
-
-	od->condition.signal(); // We are done consuming from ring_buffer
-
-	UInt32 available_frames = available_bytes / input_buffer_frame_size;
-
-	/*
-		To de-interleave the data in the input buffer so that it fits in
-		the output buffers we divide the input buffer frames into 'sub frames'
-		that fit into the output buffers.
-	*/
-	size_t sub_frame_offset = 0;
-	for (unsigned int i = 0 ; i < buffer_list->mNumberBuffers; ++i) {
-		output_buffer = &buffer_list->mBuffers[i];
-		output_buffer_frame_size = output_buffer->mNumberChannels * sample_size;
-		output_buffer->mDataByteSize = 0; // Record how much data we actually rendered
-		for (UInt32 current_frame = 0; current_frame < available_frames; ++current_frame) {
-				memcpy(
-					(uint8_t *) output_buffer->mData + current_frame * output_buffer_frame_size,
-					od->render_buffer + current_frame * input_buffer_frame_size + sub_frame_offset,
-					output_buffer_frame_size
-				);
-				output_buffer->mDataByteSize += output_buffer_frame_size;
-		}
-		sub_frame_offset += output_buffer_frame_size;
-	}
-
-	return noErr;
+	int count = in_number_frames * od->asbd.mBytesPerFrame;
+	buffer_list->mBuffers[0].mDataByteSize =
```

Comment:
How do you know there is only one output buffer? Is this documented by Apple anywhere?

To be clear, I am not asking because I think this is wrong, but because I wonder how one is supposed to find out about these things.